modules/core/arrow-test/src/main/kotlin/arrow/test/laws/AsyncLaws.kt

```diff
@@ -84,33 +83,34 @@ object AsyncLaws {
       async(k).equalUnderTheLaw(asyncF { cb -> delay { k(cb) } }, EQ)
     }
 
-  fun <F> Async<F>.bracketReleaseIscalledOnCompletedOrError(EQ: Eq<Kind<F, Int>>): Unit =
+  fun <F> Async<F>.bracketReleaseIscalledOnCompletedOrError(EQ: Eq<Kind<F, Int>>): Unit {
+    val async = this
     forAll(Gen.string().map(::just), Gen.int()) { fa, b ->
-      Promise.uncancelable<F, Int>(this).flatMap { promise ->
+      Promise.uncancelable<F, Int>(async).flatMap { promise ->
```

Comment:
Instead of doing this, you could use named parameters, such as `Promise.uncancelable<F, Int>(async = this)`